### DISCUSSION TITLE
.clang-format support?

### DISCUSSION THREAD

**Discussion by Jeordanis** [zed-industries/zed#6874] [Feature Requests]
Hey guys,

I’m trying to make clang-format work. In my settings I have:
`  "formatter": {
    "external": {
      "command": "/usr/local/bin/clang-format",
      "arguments": [
        "--style=file",
        "{buffer_path}"
      ]
    }
  },
`
This works. The problem is, not saved changes will be lost when formatting. 

Nothing happens if I use `"--stdin-file-path={buffer_path}" ` as file parameter. 

Using the clang-format argument “-i” to write the file in place works fine to format the file on disk but the buffer will not be updated, so Zed ends up with a blank buffer. 

Any ideas how to approach this so the formatting works as intended? 

Is there a way to configure formatting rules?l with the default Zed formatted?  

Thanks!

**Comment by Jeordanis** [comment]
Fixed by using the argument `”--assume-filename={buffer_path}"
`

`      "formatter": {
        "external": {
          "command": "/usr/local/bin/clang-format",
          "arguments": [
            "--style=file",
            "--assume-filename={buffer_path}"
          ]
        }
      },
`

**Comment by SpencerLommel** [comment]
For anyone trying to setup clang-format on MacOS here's a guide for it

1. Install clang-format
 - [Install Brew if not already](https://brew.sh/)
 - Install clang format `brew install clang-format`
 
2. Add config to Zed
- Open `~/.config/zed/settings.json` ("Ctrl/Command + ," shortcut to open in Zed)
- Add the following lines to ensure clang is used.
```
"formatter": {
    "external": {
      "command": "clang-format",
      "arguments": ["--style=file", "--assume-filename={buffer_path}"]
    }
  },
  "format_on_save": "on"
```

The comment by @Jeordanis specifies the full path which works as well but I had issues with it when installing clang-format with brew

Hope this helps!
